pkg/setting/setting_azure.go

```diff
@@ -69,6 +69,18 @@ func (cfg *Cfg) readAzureSettings() {
 		azureSettings.UserIdentityFallbackCredentialsEnabled = azureSection.Key("user_identity_fallback_credentials_enabled").MustBool(true)
 	}
 
+	if customCloudsJSON := azureSection.Key("clouds_config").MustString(""); customCloudsJSON != "" {		
+		if err := azureSettings.SetCustomClouds(customCloudsJSON); err != nil {
+			cfg.Logger.Error("Failed to parse custom Azure cloud settings", "err", err.Error())
+		} else {
+			// TODO REMOVE THIS ELSE_BLOCK_CODE ONCE TESTING IS DONE
+			customClouds := azureSettings.CustomClouds()
```

Comment:
I just remembered that CustomClouds returns just the AzureSettings struct (with just the name/description) but the plugin needs the the AzureCloudSettings struct (which has all the properties).  We will need to either expose the customClouds field or we will need to reconstruct the list of AzureCloudSettings list via calls to CustomClouds and GetCloud(cloudName)

Reply:
For now I'd suggest using the `GetCloud` function with the cloud name - that is how the implementation currently works in the plugins.

Reply:
The else block will go away before I submit the official PR, so no call to CustomClouds will happen.